polygon/sync/sync_to_tip_events.go

```diff
@@ -1,34 +1,135 @@
 package sync
 
 import (
+	"container/list"
+	"context"
+	"sync"
+	"time"
+
 	"github.com/ledgerwatch/erigon/core/types"
 	"github.com/ledgerwatch/erigon/polygon/heimdall"
 	"github.com/ledgerwatch/erigon/polygon/p2p"
 )
 
-const EventTypeMilestone = "milestone"
 const EventTypeNewHeader = "new-header"
+const EventTypeMilestone = "milestone"
 const EventTypeNewSpan = "new-span"
 
 type Event struct {
 	Type string
 
-	Milestone *heimdall.Milestone
-
+	// EventTypeNewHeader
 	NewHeader *types.Header
 	PeerId    p2p.PeerId
 
+	// EventTypeMilestone
+	Milestone *heimdall.Milestone
+
+	// EventTypeNewSpan
 	NewSpan *heimdall.Span
 }
 
 type SyncToTipEvents struct {
-	events chan Event
+	events    chan Event
```

Comment:
there is a convention that a struct name should not start with same name of package
consider renaming `SyncToTipEvents` just to `TipEvents`